### DISCUSSION TITLE
Accessing search query in search results column

### DISCUSSION THREAD

**Discussion by upcreek** [microsoft-search/pnp-modern-search#4157] [Q&A] [answered]
Hello. Is there anyway to access the search query from a column in the search results web part (specifically, the details list layout)? I'm looking to use an expression in a column to show a particular property based on the search query. 

E.g. If search query is X, show managed property Y, else show managed property Z.

Thanks!

**Comment by wobba** [comment]
Short answer is yes. You can get it in the template via `{{@root.inputQueryText}}`.

**Comment by upcreek** [comment]
Can that be used in a column in a handlebars expression, or is it only available in the template? (Not sure if I'm using the correct jargon here) On Dec 14, 2024 9:26 AM, Mikael Svenson ***@***.***> wrote:
Short answer is yes. You can get it in the template via ***@***.***}}.

**Comment by kasperbolarsen** [comment]
{{@root.inputQueryText}} is a handlebar so you can absolutely use it there

**Reply by upcreek** [reply]
Is there any special configuration needed to expose the query? Seems like no matter where I put that handlebars expression, I get nothing back. I must be missing something. Is it only available on certain layout templates?

**Reply by wobba** [reply]
In all templates. Pick your starting template from the default ones. Then click the `{}` button which allows you to customize the one you started with. And add the expression somewhere below the first div like:

```
<div class="template">
{{@root.inputQueryText}}
```

**Reply by upcreek** [reply]
Ah ha, a bit closer. This works in a custom template, but I can't get it to work in the handlebars expression of a column in the details list template. What I'm ultimately going for is being able to use the query value to do some logic on a column value in each search result. Would I have to do a complete custom template for this?

**Reply by wobba** [reply]
Good question and something to look into indeed.. why that token cannot be referenced in a column expression.

**Reply by upcreek** [reply]
At least I know it's not me. I'll adjust and move on - it's not critical. Thanks for your time!

**Comment by wobba** [comment]
Should be fixed with https://github.com/microsoft-search/pnp-modern-search/pull/4177

**Comment by jblueml** [comment, marked as answer]
@upcreek : Hi, with the latest version 4.15.0 of PnP Modern Search, you should now be able to reference `{{@root.inputQueryText}}` in the handlebar expression of a Detailslist-column. Could you please check if it now works as expected?

**Reply by upcreek** [reply]
Works perfect! Thanks!
